src/index.js

```diff
@@ -0,0 +1,213 @@
+      updateCompleted(event, task, listText);
+      localStorage.setItem('tasks', JSON.stringify(tasks));
+    });
+  });
+};
+orderTasks();
+generateTasks();
+
+// start addRemove js
+
+const addNewTask = () => {
+  const inputAdd = document.querySelector('#tasks-text').value;
+  let newIndex;
+  if (tasks === null) {
+    newIndex = 1;
+  } else {
+    newIndex = tasks.length + 1;
+  }
+  const task = {
+    description: inputAdd,
+    index: newIndex,
+    completed: false,
+  };
+  tasks.push(task);
+  localStorage.setItem('tasks', JSON.stringify(tasks));
+  return task;
+};
+const generateOneTask = (task) => {
+  const div1 = document.createElement('div');
+  const div2 = document.createElement('div');
+  const squareIcon = document.createElement('input');
+  squareIcon.type = 'checkbox';
+  const listText = document.createElement('input');
+  listText.type = 'text';
+  const threeDotIcon = document.createElement('i');
+  div1.appendChild(div2);
+  div1.appendChild(threeDotIcon);
+  div2.appendChild(squareIcon);
+  div2.appendChild(listText);
+
+  div1.className = 'list-items';
+  div2.className = 'left-items';
+  listText.className = 'list-text1';
+  squareIcon.className = 'checkbox';
+  threeDotIcon.className = 'fas fa-ellipsis-v three-dot-icon';
+  document.querySelector('.box').appendChild(div1);
+  listText.value = task.description;
+
+  const deleteIcon = document.createElement('i');
+  deleteIcon.className = 'fas fa-trash';
+  listText.addEventListener('keyup', () => {
+    task.description = listText.value;
+    localStorage.setItem('tasks', JSON.stringify(tasks));
+  });
+  listText.addEventListener('click', () => {
+    threeDotIcon.replaceWith(deleteIcon);
+  });
+  listText.addEventListener('blur', () => {
+    setTimeout(() => {
+      deleteIcon.replaceWith(threeDotIcon);
+    }, 300);
+  });
+
+  deleteIcon.addEventListener('click', (event) => {
+    tasks = JSON.parse(localStorage.getItem('tasks'));
+    const id = task.index - 1;
+    tasks = tasks.filter((task, ind) => ind !== id);
+    localStorage.setItem('tasks', JSON.stringify(tasks));
+    updatedIndex();
+    event.target.parentNode.remove();
+  });
+
+  // Clear all completed
+  const clearSelected = (tasks) => {
+    tasks = JSON.parse(localStorage.getItem('tasks'));
+    tasks = tasks.filter((task) => !task.completed);
+    localStorage.setItem('tasks', JSON.stringify(tasks));
+    return true;
+  };
+  const clear = document.querySelector('.complete-text');
+  clear.addEventListener('click', () => {
+    clearSelected(tasks);
+    updatedIndex();
+  });
+
+  if (task.completed === true) {
+    listText.classList.add('list-text');
+    squareIcon.checked = true;
+  } else {
+    listText.classList.remove('list-text');
+    squareIcon.checked = false;
+  }
+  squareIcon.addEventListener('change', (event) => {
+    updateCompleted(event, task, listText);
+    localStorage.setItem('tasks', JSON.stringify(tasks));
+  });
+};
```

Comment:
- [ ] Refactor this code so that its implementation is not identical to the one on line 35